apm-agent-core/src/main/java/co/elastic/apm/agent/logging/Log4j2LoggerBridge.java

```diff
@@ -44,151 +47,235 @@ public String getName() {
-        log4jLogger.logIfEnabled(FQCN, Level.INFO, null, format, arg1, arg2);
+        logIfEnabled(Level.INFO, format, arg1, arg2);
     }
 
     @Override
     public void info(final String format, final Object... args) {
-        log4jLogger.logIfEnabled(FQCN, Level.INFO, null, format, args);
+        logIfEnabled(Level.INFO, format, args);
     }
 
     @Override
     public void info(final String format, final Throwable t) {
-        log4jLogger.logIfEnabled(FQCN, Level.INFO, null, format, t);
+        logIfEnabled(Level.INFO, format, t);
     }
 
     @Override
     public boolean isWarnEnabled() {
-        return log4jLogger.isEnabled(Level.WARN, null, null);
+        return isEnabled(Level.WARN);
     }
 
     @Override
     public void warn(final String format) {
-        log4jLogger.logIfEnabled(FQCN, Level.WARN, null, format);
+        logIfEnabled(Level.WARN, format);
     }
 
     @Override
     public void warn(final String format, final Object o) {
-        log4jLogger.logIfEnabled(FQCN, Level.WARN, null, format, o);
+        logIfEnabled(Level.WARN, format, o);
     }
 
     @Override
     public void warn(final String format, final Object arg1, final Object arg2) {
-        log4jLogger.logIfEnabled(FQCN, Level.WARN, null, format, arg1, arg2);
+        logIfEnabled(Level.WARN, format, arg1, arg2);
     }
 
     @Override
     public void warn(final String format, final Object... args) {
-        log4jLogger.logIfEnabled(FQCN, Level.WARN, null, format, args);
+        logIfEnabled(Level.WARN, format, args);
     }
 
     @Override
     public void warn(final String format, final Throwable t) {
-        log4jLogger.logIfEnabled(FQCN, Level.WARN, null, format, t);
+        logIfEnabled(Level.WARN, format, t);
     }
 
     @Override
     public boolean isErrorEnabled() {
-        return log4jLogger.isEnabled(Level.ERROR, null, null);
+        return isEnabled(Level.ERROR);
     }
 
     @Override
     public void error(final String format) {
-        log4jLogger.logIfEnabled(FQCN, Level.ERROR, null, format);
+        logIfEnabled(Level.ERROR, format);
     }
 
     @Override
     public void error(final String format, final Object o) {
-        log4jLogger.logIfEnabled(FQCN, Level.ERROR, null, format, o);
+        logIfEnabled(Level.ERROR, format, o);
     }
 
     @Override
     public void error(final String format, final Object arg1, final Object arg2) {
-        log4jLogger.logIfEnabled(FQCN, Level.ERROR, null, format, arg1, arg2);
+        logIfEnabled(Level.ERROR, format, arg1, arg2);
     }
 
     @Override
     public void error(final String format, final Object... args) {
-        log4jLogger.logIfEnabled(FQCN, Level.ERROR, null, format, args);
+        logIfEnabled(Level.ERROR, format, args);
     }
 
     @Override
     public void error(final String format, final Throwable t) {
-        log4jLogger.logIfEnabled(FQCN, Level.ERROR, null, format, t);
+        logIfEnabled(Level.ERROR, format, t);
+    }
+
```

Comment:
[for reviewer] calls the the logger might open files for writing when 1) creating the log file, or 2) rotating the log file, thus we have to wrap all logger calls with elevated privileges.